server/http/interceptors/interceptors.go

```diff
@@ -37,15 +38,41 @@ import (
 
 var (
 	upgradeInsecure = flag.Bool("ssl.upgrade_insecure", false, "True if http requests should be redirected to https. Assumes http traffic is served on port 80 and https traffic is served on port 443 (typically via an ingress / load balancer).")
-
-	uuidV4Regexp = regexp.MustCompile("[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}")
 )
 
+const ContentSecurityPolicyHeader = "Content-Security-Policy"
+
+func GetContentSecurityPolicy(scriptHashes ...[sha256.Size]byte) string {
+	scriptSrcs := make([]string, 0, len(scriptHashes))
+	for _, scriptHash := range scriptHashes {
+		scriptSrcs = append(scriptSrcs, fmt.Sprintf("'sha256-%s'", base64.StdEncoding.EncodeToString(scriptHash[:])))
+	}
+	return strings.Join([]string{
+		"default-src 'self'",
+		"style-src 'self' https://fonts.googleapis.com/css",
+		// The fixed hash is for the inline script that injects the Google Tag Manager. Additional hashes can be
+		// added by the caller.
+		"script-src 'self' https://www.googletagmanager.com 'sha256-3dO3w7Y+Pvr8nNuM9JioBpuBGrxgMBNZhPSqYDD4qn4=' " + strings.Join(scriptSrcs, " "),
```

Comment:
nit: wdyt about using a [nonce-based](https://web.dev/articles/strict-csp#structure) approach instead of hash-based? I am not an expert in CSP but it seems like nonce-based might be a little easier to maintain since we don't have to update these hashes if the contents change

Reply:
A hash-based approach generally has slightly better security guarantees (think `</script>` injection into server-generated data), but they don't apply to our current usage. It also makes the page itself static, which can help with caching.

A nonce-based approach doesn't require updating the hashes, which are currently only two which are probably very stable, but you are right that this could turn out to be more maintainable in the future. 

It's a close call for me. I did update the PR to use the nonce-based approach since it makes it easier for users to customize the index page: they just need to add `{{.Nonce}}` to their `script` tags rather than having to adding hashes to the Go constant.